src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -160,44 +211,52 @@ class BatchNorm : public Layer<InputType, OutputType>
   //! Locally-stored value for momentum.
   double momentum;
 
-  //! Variable to keep track of whether we are in loading or saving mode.
-  bool loading;
-
   //! Locally-stored scale parameter.
-  OutputType gamma;
+  MatType gamma;
 
   //! Locally-stored shift parameter.
-  OutputType beta;
-
-  //! Locally-stored mean object.
-  OutputType mean;
+  MatType beta;
 
   //! Locally-stored variance object.
-  OutputType variance;
+  MatType variance;
 
   //! Locally-stored parameters.
-  OutputType weights;
+  MatType weights;
 
   //! Locally-stored running mean/variance counter.
   size_t count;
 
-  //! Locally-stored value for average factor which used to update running
-  //! mean and variance.
-  double averageFactor;
+  //! Locally-stored number of input dimensions that we are applying 
+  //! batch normalization over.  (This is the product of this->inputDimensions 
+  //! from index 0 to (maxAxis - 1)).
```

Comment:
```suggestion
  //! from index 0 to (minAxis - 1)).
```
I think this is `minAxis` now?